src/arrayMethodReduce.js

```diff
@@ -5,7 +5,17 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+
+    if (arguments.length < 2) {
+      accumulator = this[0];
```

Comment:
You could declare i = 0 and change it to 1 in this if block. It would make your loop cleaner